src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -99,17 +99,47 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
             case 2:
                 genConsumeOperands(node);
+                op1Reg = op1->gtRegNum;
+                op2Reg = op2->gtRegNum;
                 if (category == HW_Category_MemoryStore)
                 {
-                    emit->emitIns_AR_R(ins, simdSize, op2->gtRegNum, op1->gtRegNum, 0);
+                    emit->emitIns_AR_R(ins, simdSize, op2Reg, op1Reg, 0);
                 }
                 else if ((ival != -1) && varTypeIsFloating(baseType))
                 {
                     genHWIntrinsic_R_R_RM_I(node, ins);
                 }
                 else if (category == HW_Category_MemoryLoad)
                 {
-                    emit->emitIns_SIMD_R_R_AR(ins, simdSize, targetReg, op1->gtRegNum, op2->gtRegNum);
+                    emit->emitIns_SIMD_R_R_AR(ins, simdSize, targetReg, op1Reg, op2Reg);
+                }
+                else if (Compiler::isImmHWIntrinsic(intrinsicID, op2))
+                {
+                    auto emitSwCase = [&](int i) {
+                        if ((ins == INS_psrldq || ins == INS_pslldq) && i > 15)
```

Comment:
This special case can/should be moved to `emitIns_SIMD_R_R_I`.

Reply:
Will change.